### PR TITLE
[FIX] purchase_last_price_info: multicompany error, closes #1406

### LINKED ISSUE #1406: [14.0] purchase_last_price_info not working on multi-company environment with shared products
## Module
purchase_last_price_info

## Describe the bug
On multi-company environment with shared products:
- Buy PRODUCT1 on CompanyA => purchase.line.order belongs to CampanyA
- Go to CompanyB => Inventory => Products (kanban) => Change to tree view (ERROR because last_price_info is listed):

Access Error
Due to security restrictions, you are not allowed to access 'Purchase Order Line' (purchase.order.line) records.
Records: [FURN_6666] Acoustic Bloc Screens (id=19)
User: Mitchell Admin (id=2)
This restriction is due to the following rules:
- Purchase Order Line multi-company
Note: this might be a multi-company issue.
Contact your administrator to request access if necessary.

## To Reproduce
**14.0**:

**Expected behavior**
The module must search the purchase.order.line on the current company.

**Additional context**
Tested on runboat with demo data

### PR DESCRIPTION
**Pull request by dreispt**

### CODE REVIEW

File: purchase_last_price_info/models/product_product.py [hunk 1]
```diff
@@ -12,7 +12,8 @@ class ProductProduct(models.Model):
     last_purchase_line_ids = fields.One2many(
         comodel_name="purchase.order.line",
         inverse_name="product_id",
-        domain=[("state", "in", ["purchase", "done"])],
+        domain="[('state', 'in', ['purchase', 'done']),"
+        "('company_id', 'in', company_ids)]",
```
Comment (ivantodorovich) [inline purchase_last_price_info/models/product_product.py:16]: Why is this needed?
Isn't it handled by `purchase.order.line` ir rules?

Comment (dreispt) [reply purchase_last_price_info/models/product_product.py:16]: Indeed, but It doesn't look so, or we would not see the bug.

**Comment by patrickrwilson** [comment]
@dreispt tested this with runboat but still see the error. 
Created PR in My Company (San Francisco), validated and received products. Switched to the other company, went to inventory->products and click on list view then the following shows...
image

File: purchase_last_price_info/models/product_product.py (hunk 1 above)
Comment (ghost) [reply purchase_last_price_info/models/product_product.py:16]: that solution isn't work for me...

this solution work fine:
domain=lambda self: [('state', 'in', ['purchase', 'done']),
                             ('company_id', '=', self.env.company.id)]

**Comment by pedrobaeza** [comment]
I have pushed the proper patch (as @AladroEticco has proposed) in 2d8ed069aab225161e40ab304c479f56e17ba1fc, so closing this. Thanks all for the involvement.

**Comment by dreispt** [comment]
Thank you @pedrobaeza, can you share the link to the corresponding PR please?

**Comment by pedrobaeza** [comment]
It's a direct commit, not a PR. As maintainer, I have done such quick flow.

**Comment by dreispt** [comment]
I don't think this is proper OCA process. Have I missed a discussion about it  somewhere?

**Comment by pedrobaeza** [comment]
I ask you the contrary: tell me where in the guidelines is forbidden, and don't be nitpicking with someone that knows what is doing (or fix whatever necessary, even not directly provoked by him) and it's doing hundreds of pull requests (when needed). This unlocks this specific situation and get a satisfactory end as one line commit, so it's OK to take as maintainer this quick flow. It has been done a lot of times on maintenance operations or for solving critical bugs.

**Comment by dreispt** [comment]
IMO it goes against the principle where code merged must have a reviewer, especially for stable/mature modules.
Even if fast-tracking is needed, a PR should be there, and opportunity for review, or for other people to be aware of the merge.

Only exceptions I've seen to this are technical repo maintenance commits, that don't hurt the review principles since they don't affect actual module code.

Maybe this should be discussed and clarified.
